pkg/util/inplaceupdate/inplace_utils.go

```diff
@@ -0,0 +1,86 @@
+package inplaceupdate
+
+import (
+	"encoding/json"
+	"errors"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	appspub "github.com/openkruise/kruise/apis/apps/pub"
+	"github.com/openkruise/kruise/pkg/util"
+	"github.com/openkruise/kruise/pkg/util/metrics/update"
+	"github.com/openkruise/kruise/pkg/util/podadapter"
+)
+
+const AnnotationRestartCountInfoKey = "inplace.apps.kruise.io/restart-count-info"
+
+type RestartCountInfo struct {
+	RestartCountBase map[string]int32 `json:"base"`
+}
+
+func (c *realControl) RefreshRestartCountBaseToPod(pod *v1.Pod) (gotPod *v1.Pod, err error) {
+	if len(pod.Status.ContainerStatuses) == 0 {
+		return pod, errors.New("no container status")
+	}
+	base := &RestartCountInfo{
+		RestartCountBase: make(map[string]int32),
+	}
+	for _, cs := range pod.Status.ContainerStatuses {
+		base.RestartCountBase[cs.Name] = cs.RestartCount
+	}
+
+	info := util.DumpJSON(base)
+	podCopy := pod.DeepCopy()
```

Comment:
it is not necessary to deepcopy the pod, just initiate an empty pod with the given pod name/namespace